x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/rest/action/RestPreviewTransformAction.java

```diff
@@ -34,9 +37,27 @@ public String getName() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
-        XContentParser parser = restRequest.contentParser();
-
-        PreviewTransformAction.Request request = PreviewTransformAction.Request.fromXContent(parser);
+        boolean hasId = restRequest.hasParam(TransformField.ID.getPreferredName());
+        boolean hasContent = restRequest.hasContent();
+        PreviewTransformAction.Request request;
+        if (hasId && hasContent) {
+            String idFromParam = restRequest.param(TransformField.ID.getPreferredName());
+            String idFromBody = (String) restRequest.contentParser().map().get(TransformField.ID.getPreferredName());
+            if (idFromParam.equals(idFromBody) == false) {
+                throw new ElasticsearchParseException(
+                    "transform id param [" + idFromParam + "] does not match request body id [" + idFromBody + "]");
+            }
+            XContentParser parser = restRequest.contentParser();
+            request = PreviewTransformAction.Request.fromXContent(parser);
+        } else if (hasId) {
```

Comment:
This seems way too complicated. I don't see the benefit. Either the user is checking a transform that already exists or they are not.

Besides, for `preview` we don't care about the ID provided in the body.

Reply:
Take a look at how Data frame analytics handles this: 

https://github.com/elastic/elasticsearch/blob/8384c0d67816758621aa3b595f8fd2a3b2521cfc/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/dataframe/RestPreviewDataFrameAnalyticsAction.java#L28